source/renderer/app/api/api.js

```diff
@@ -548,32 +548,45 @@ export default class AdaApi {
           },
         ],
       };
+
       const response: TransactionFee = await getTransactionFee(this.config, {
         walletId,
         data,
       });
 
+      const formattedTxAmount = new BigNumber(request.amount).dividedBy(
+        LOVELACES_PER_ADA
+      );
+      const fee = _createTransactionFeeFromServerData(response);
+      const amountWithFee = formattedTxAmount.plus(fee);
+      if (amountWithFee.gt(walletBalance)) {
+        // Amount + fees exceeds walletBalance:
+        // = show "Not enough Ada for fees. Try sending a smaller amount."
+        throw new NotEnoughFundsForTransactionFeesError();
+      }
+
       Logger.debug('AdaApi::calculateTransactionFee success', {
         transactionFee: response,
       });
-
-      return _createTransactionFeeFromServerData(response);
+      return fee;
     } catch (error) {
       Logger.error('AdaApi::calculateTransactionFee error', { error });
-      if (error.code === 'not_enough_money') {
-        // 1 utxo available
-        // created tx with one output
-        // transaction amount is greater than wallet's balance
-        throw new NotEnoughFundsForTransactionError();
-      } else if (error.code === 'utxo_not_enough_fragmented') {
-        // 1 utxo available
-        // created tx with 2 outputs
-        // calculated fees PLUS the transaction amount exceeds wallet's balance
-        // transaction amount is less than wallet's balance
-
-        // @API TODO - Change error message when fee calculation fails regarding to not enough fragmented UTXO
-        //           - Also check if error.code is correct
-        throw new NotEnoughFundsForTransactionError();
+      if (error.name === 'NotEnoughFundsForTransactionFeesError') {
+        throw new NotEnoughFundsForTransactionFeesError();
+      } else if (error.code === 'not_enough_money') {
+        if (request.walletBalance.gt(request.availableBalance)) {
```

Comment:
@tomothespian You can just use `if (walletBalance.gt(availableBalance)) {` here.